buffer/src/test/java/io/netty/buffer/AbstractByteBufTest.java

```diff
@@ -2229,20 +2248,25 @@ public void run() {
         latch.await(10, TimeUnit.SECONDS);
         barrier.await(5, TimeUnit.SECONDS);
         assertNull(cause.get());
+        buffer.release();
     }
 
     @Test(expected = IndexOutOfBoundsException.class)
     public void readByteThrowsIndexOutOfBoundsException() {
-        final ByteBuf buffer = releaseLater(newBuffer(8));
-        buffer.writeByte(0);
-        assertEquals((byte) 0, buffer.readByte());
-        buffer.readByte();
+        final ByteBuf buffer = newBuffer(8);
+        try {
+            buffer.writeByte(0);
+            assertEquals((byte) 0, buffer.readByte());
+            buffer.readByte();
+        } finally {
```

Comment:
@Scottmitch actually now I remember why I even put these in finally blocks. Its because all of these tests will throw an exception and if we not put it in a finally block we will leak. So ok to leave these ?